airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -433,10 +433,38 @@ def post_clear_task_instances(*, dag_id: str, session: Session = NEW_SESSION) ->
     if not dag:
         error_message = f"Dag id {dag_id} not found"
         raise NotFound(error_message)
-    reset_dag_runs = data.pop('reset_dag_runs')
-    dry_run = data.pop('dry_run')
-    # We always pass dry_run here, otherwise this would try to confirm on the terminal!
-    task_instances = dag.clear(dry_run=True, dag_bag=current_app.dag_bag, **data)
+    reset_dag_runs = data.pop('reset_dag_runs') == "true"
```

Comment:
There is a `to_boolean` function we can use to better parse this.

Reply:
i will check this @uranusjr